src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1588,8 +1604,14 @@ InferenceProfiler::SummarizeServerStatsHelper(
 
   const auto& end_itr = end_status.find(this_id);
   if (end_itr == end_status.end()) {
-    return cb::Error(
-        "missing statistics for requested model", pa::GENERIC_ERROR);
+    if (parser_ == nullptr || !parser_->TopLevelResponseCachingEnabled()) {
```

Comment:
Same comment about assigning a meaningful name to this combination (or a comment)